src/flagpole/conditions.py

```diff
@@ -1,25 +1,202 @@
+                f"'In' condition value must be a list, but was provided a '{get_type_name(self.value)}'"
+                + f" of segment {segment_name}"
+            )
+        if isinstance(condition_property, (list, dict)):
+            raise ConditionTypeMismatchException(
+                "'In' condition property value must be str | int | float | bool | None, but was provided a"
+                + f"'{get_type_name(self.value)}' of segment {segment_name}"
+            )
+        if isinstance(condition_property, str):
+            condition_property = condition_property.lower()
+
+        return condition_property in create_case_insensitive_set_from_list(self.value)
+
+    def _evaluate_contains(self, condition_property: Any, segment_name: str) -> bool:
+        if not isinstance(condition_property, list):
+            raise ConditionTypeMismatchException(
+                f"'Contains' can only be checked against a list, but was given a {get_type_name(condition_property)}"
+                + f" context property '{condition_property}' of segment '{segment_name}'"
+            )
+        value = self.value
+        if isinstance(value, str):
+            value = value.lower()
+
+        return value in create_case_insensitive_set_from_list(condition_property)
+
+    def _evaluate_equals(self, condition_property: Any, segment_name: str) -> bool:
+        if not isinstance(condition_property, type(self.value)):
+            value_type = get_type_name(self.value)
+            property_value = get_type_name(condition_property)
+            raise ConditionTypeMismatchException(
+                "'Equals' operator cannot be applied to values of mismatching types"
+                + f"({value_type} and {property_value}) for segment {segment_name}"
+            )
+
+        if isinstance(condition_property, str):
+            return condition_property.lower() == self.value.lower()
+
+        return condition_property == self.value
+
+
+InOperatorValueTypes = list[StrictInt] | list[StrictFloat] | list[StrictStr]
+
+
+class InCondition(ConditionBase):
+    operator: Literal[ConditionOperatorKind.IN] = ConditionOperatorKind.IN
+    value: InOperatorValueTypes
+
+    def _operator_match(self, condition_property: Any, segment_name: str):
+        return self._evaluate_in(condition_property=condition_property, segment_name=segment_name)
+
+
+class NotInCondition(ConditionBase):
+    operator: Literal[ConditionOperatorKind.NOT_IN] = ConditionOperatorKind.NOT_IN
+    value: InOperatorValueTypes
+
+    def _operator_match(self, condition_property: Any, segment_name: str):
+        return not self._evaluate_in(
+            condition_property=condition_property, segment_name=segment_name
+        )
+
+
+ContainsOperatorValueTypes = StrictInt | StrictStr | StrictFloat
+
+
+class ContainsCondition(ConditionBase):
+    operator: Literal[ConditionOperatorKind.CONTAINS] = ConditionOperatorKind.CONTAINS
+    value: StrictInt | StrictStr | StrictFloat
```

Comment:
```suggestion
    value: ContainsOperatorValueTypes
```

This should be equivalent.